pkg/volume/gcepd/attacher.go

```diff
@@ -384,9 +384,26 @@ func (detacher *gcePersistentDiskDetacher) Detach(volumeName string, nodeName ty
 }
 
 func (detacher *gcePersistentDiskDetacher) UnmountDevice(deviceMountPath string) error {
+	if runtime.GOOS == "windows" {
+		// Flush data cache for windows because it does not do so automatically during unmount device
+		exec := detacher.host.GetExec(gcePersistentDiskPluginName)
+		err := writeVolumeCache(deviceMountPath, exec)
+		if err != nil {
+			return err
+		}
+	}
 	return mount.CleanupMountPoint(deviceMountPath, detacher.host.GetMounter(gcePersistentDiskPluginName), false)
 }
 
+func writeVolumeCache(deviceMountPath string, exec mount.Exec) error {
```

Comment:
Since this is not GCE PD specific and applies to various other volume plugins such EBS, AzureDisk, iSCSI, etc. how about adding this to [volume/util/util](https://github.com/kubernetes/kubernetes/blob/8a106a57146b00b1d568fb170e87f19af238d2f4/pkg/volume/util/util.go)?

Reply:
fixed. Thanks for the advice